dotnet/src/Functions/Functions.OpenApi/Model/RestApiOperation.cs

```diff
@@ -275,21 +303,34 @@ private Uri GetServerUrl(Uri? serverUrlOverride, Uri? apiHostUrl, IDictionary<st
         else if (this.Servers is { Count: > 0 } servers && servers[0].Url is { } url)
         {
             serverUrlString = url;
+
             foreach (var variable in servers[0].Variables)
             {
-                arguments.TryGetValue(variable.Key, out object? value);
-                string? strValue = value as string;
-                if (strValue is not null && variable.Value.IsValid(strValue))
+                var variableName = variable.Key;
+
+                // Try to get the variable value by the argument name.
+                if (!string.IsNullOrEmpty(variable.Value.ArgumentName) &&
+                    arguments.TryGetValue(variable.Value.ArgumentName!, out object? value) &&
+                    value is string { } argStrValue && variable.Value.IsValid(argStrValue))
+                {
+                    serverUrlString = url.Replace($"{{{variableName}}}", argStrValue);
+                }
+                // Try to get the variable value by the variable name.
+                else if (arguments.TryGetValue(variableName, out value) &&
+                    value is string { } strValue &&
+                    variable.Value.IsValid(strValue))
                 {
-                    serverUrlString = serverUrlString.Replace($"{{{variable.Key}}}", strValue);
+                    serverUrlString = url.Replace($"{{{variableName}}}", strValue);
                 }
+                // Use the default value if no argument is provided.
                 else if (variable.Value.Default is not null)
                 {
-                    serverUrlString = serverUrlString.Replace($"{{{variable.Key}}}", variable.Value.Default);
+                    serverUrlString = url.Replace($"{{{variableName}}}", variable.Value.Default);
                 }
+                // Throw an exception if there's no value for the variable.
                 else
                 {
-                    throw new KernelException($"No value provided for the '{variable.Key}' server variable of the operation - '{this.Id}'.");
+                    throw new KernelException($"No argument provided for the '{variableName}' server variable of the operation - '{this.Id}'.");
```

Comment:
Same as above it would be useful to know what argument name was tried if one is set

Reply:
done